tools/windows-node-installer/pkg/cloudprovider/azure/vm.go

```diff
@@ -80,6 +81,8 @@ type AzureProvider struct {
 	nsgClient network.SecurityGroupsClient
 	// diskClient to query for disk related operations.
 	diskClient compute.DisksClient
+	// resourcesClient to query resources
```

Comment:
which resources specifically are you interested in, in this scenario?

Reply:
resources client is used to query all resources available. 
i'm querying vm resources using the resources client.